tutorials/misc/plot_ecog.py

```diff
@@ -183,20 +166,118 @@ def animate(i, activity):
+# very similarly to the high gamma visualization.
+#
+# As can be seen in the plot below, the seizure originates in the temporal
+# lobe and spreads to new electrodes when electrode label names come up in
+# the title. The seizure eventually becomes generalized and then subsides.
+
+# colormap to view spectral power - RdBu_r works well for directional
+# data such as voltage time series, where positive and negative values
+# matter and there is a true 0 point
+cmap = 'RdBu_r'
+
+# Apply a notch filter to remove 60 Hz line noise and harmonics
+raw_notched = raw.copy().pick_types(
+    meg=False, ecog=True).notch_filter([60, 120])
+
+# Apply a gentle high pass filter to get rid of drift
+raw_notched.filter(l_freq=0.1, h_freq=None)
+
+# Downsample the data to make the animation more reasonable
+sfreq = 50  # Hz
+raw_notched.resample(sfreq)
+ts_data = raw_notched.get_data()
+
+# Find the annotated events
+events, event_id = mne.events_from_annotations(raw)
+inv_event_id = {v: k for k, v in event_id.items()}
+
+# find the onset event, use the one second before as the animation start
+onset_events = events[events[:, 2] == event_id['onset']]
+start_sample = int(onset_events[0, 0] - 1 * raw.info['sfreq'])
+
+
+# Create an initialization and animation function to pass to FuncAnimation.
+# This time we will also pass the plot title so that can be updated
+# with information from the annotations.
+def init():
+    """Create an empty frame."""
+    return paths, title
+
+
+def animate(i, activity, events):
+    """Animate the plot."""
+    paths.set_array(activity[:, i])
+    tsl2 = slice(i + start_sample - sfreq, i + start_sample + sfreq)
+    for j, tpath in enumerate(tpaths):
+        tpath.set_data(raw_notched.times[tsl], ts_data[j, tsl2])
+    # If this sample contains an annotation (for a seizure-related behavior)
+    # then change the title of the plot
+    if i + start_sample in events[:, 0]:
+        # Currently this doesn't replace the text, but writes over it.
+        # This needs fixing
+        event_idx = np.argwhere(events[:, 0] == (i + start_sample))[0][0]
+        title.set_text(inv_event_id[events[event_idx, 2]])
+        fig.canvas.draw()  # force redrawing
+    return paths, title
+
+
+# create the figure and apply the animation of the
+# raw time series data
+fig, ax = plt.subplots(figsize=(5, 5))
+tax = fig.add_axes([0.15, 0.02, 0.6, 0.15])
+ax.imshow(im)
+ax.set_axis_off()
+tax.set_axis_off()
+# We want the mid-point (0 uV) to be white, so we will scale from -vmax to vmax
+# so that negative voltages are blue and positive voltages are red
+ts_data = raw.get_data()
+vmax = np.percentile(ts_data, 90)
+paths = ax.scatter(*xy_pts.T, c=np.zeros(len(xy_pts)), s=40,
+                   cmap=cmap, vmin=-vmax, vmax=vmax)
+ax.set_xlim([0, im.shape[0]])
+ax.set_ylim([im.shape[1], 0])
+tsl = slice(start_sample - sfreq, start_sample + sfreq)
+tpaths = tax.plot(raw_notched.times[tsl], ts_data[:, tsl].T)
```

Comment:
after fixing the notch filter freqs I'm still getting errors on this line when running locally: 

```
Traceback (most recent call last):
  File "/opt/mne/python/tutorials/misc/plot_ecog.py", line 248, in <module>
    tpaths = tax.plot(raw_notched.times[tsl], ts_data[:, tsl].T)
  File "/opt/miniconda3/envs/mnedev/lib/python3.8/site-packages/matplotlib/axes/_axes.py", line 1743, in plot
    lines = [*self._get_lines(*args, data=data, **kwargs)]
  File "/opt/miniconda3/envs/mnedev/lib/python3.8/site-packages/matplotlib/axes/_base.py", line 273, in __call__
    yield from self._plot_args(this, kwargs)
  File "/opt/miniconda3/envs/mnedev/lib/python3.8/site-packages/matplotlib/axes/_base.py", line 399, in _plot_args
    raise ValueError(f"x and y must have same first dimension, but "
ValueError: x and y must have same first dimension, but have shapes (0,) and (100, 86)
```

Reply:
Used the wrong `raw` vs `raw_notched`. It should be fixed now.